core/src/banking_stage.rs

```diff
@@ -116,37 +116,57 @@ impl BankingStage {
         Ok(())
     }
 
-    fn forward_buffered_packets(
+    fn handle_buffered_packets(
         socket: &std::net::UdpSocket,
         poh_recorder: &Arc<Mutex<PohRecorder>>,
         cluster_info: &Arc<RwLock<ClusterInfo>>,
-        buffered_packets: &[(SharedPackets, usize)],
-    ) -> bool {
-        if buffered_packets.is_empty() {
-            return false;
-        }
-
+        buffered_packets: &[(SharedPackets, usize, Vec<u8>)],
+    ) -> Result<UnprocessedPackets> {
         let rcluster_info = cluster_info.read().unwrap();
 
         // If there's a bank, and leader is available, forward the buffered packets
         // or, if the current node is not the leader, forward the buffered packets
-        let forward = match poh_recorder.lock().unwrap().bank() {
-            Some(_) => rcluster_info.leader_data().is_some(),
+        let (forward, consume) = match poh_recorder.lock().unwrap().bank() {
+            Some(_) => (false, rcluster_info.leader_data().is_some()),
             None => rcluster_info
                 .leader_data()
-                .map(|x| x.id != rcluster_info.id())
-                .unwrap_or(false),
+                .map(|x| (x.id != rcluster_info.id(), x.id == rcluster_info.id()))
```

Comment:
this is kind of clunky, how about?
`let is_leader = (x.id == rcluster_info.id()); (is_leader, !is_leader)`